python/ray/experimental/packaging/load_package.py

```diff
@@ -0,0 +1,146 @@
+                 runtime_env: dict):
+        self._name = name
+        self._description = desc
+        self._stub_file = stub_file
+        self._runtime_env = runtime_env
+
+        if not os.path.exists(stub_file):
+            raise ValueError("Stub file does not exist: {}".format(stub_file))
+
+        spec = importlib.util.spec_from_file_location(self._name,
+                                                      self._stub_file)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        self._module = module
+
+        for symbol in dir(self._module):
+            if not symbol.startswith("_"):
+                value = getattr(self._module, symbol)
+                if (isinstance(value, ray.remote_function.RemoteFunction)
+                        or isinstance(value, ray.actor.ActorClass)):
+                    # TODO(ekl) use the runtime_env option here instead of
+                    # the override env vars. Currently this doesn't work since
+                    # there is no way to define per-task job config.
+                    setattr(
+                        self,
+                        symbol,
+                        value.options(override_environment_variables={
+                            "RAY_RUNTIME_ENV_FILES": runtime_env["files"]
+                        }))
+                else:
+                    setattr(self, symbol, value)
+
+    def __repr__(self):
+        return "ray._RuntimePackage(module={}, runtime_env={})".format(
+            self._module, self._runtime_env)
+
+
+def _download_from_github_if_needed(config_path: str) -> str:
+    # TODO(ekl) support github URIs for the config.
+    return config_path
+
+
+def load_package(config_path: str) -> _RuntimePackage:
+    if not os.path.exists(config_path):
+        raise ValueError("Config file does not exist: {}".format(config_path))
+
+    if not ray.is_initialized():
+        # TODO(ekl) lift this requirement?
+        raise RuntimeError("Ray must be initialized first to load packages.")
+
+    config_path = _download_from_github_if_needed(config_path)
+    # TODO(ekl) validate schema?
+    config = yaml.safe_load(open(config_path).read())
+    base_dir = os.path.abspath(os.path.dirname(config_path))
+    runtime_env = config["runtime_env"]
+
+    # Autofill working directory by uploading to GCS storage.
+    if "files" not in runtime_env:
+        pkg_name = runtime_support.get_project_package_name(
+            working_dir=base_dir, modules=[])
+        pkg_uri = runtime_support.Protocol.GCS.value + "://" + pkg_name
+        if not runtime_support.package_exists(pkg_uri):
+            tmp_path = "/tmp/ray/_tmp{}".format(pkg_name)
+            runtime_support.create_project_package(
+                working_dir=base_dir, modules=[], output_path=tmp_path)
+            # TODO(ekl) does this get garbage collected correctly with the
+            # current job id?
```

Comment:
@iycheng I assume marking this package for deletion with the current job is possible after your GC PR?